### PR TITLE
modules: Add nrf 802.15.4 radio drv buffer config

### PR DESCRIPTION
**Pull request by JakubPegza**
Signed-off-by: Pegza, Jakub <Jakub.Pegza@nordicsemi.no>

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ manifest:
       revision: f1fa8241f8786198ba41155413243de36ed878a5
       path: modules/hal/infineon
     - name: hal_nordic
-      revision: 6038f912fd34fdd46dbf017057908f979e298341
+      revision: pull/29443/head
```
Comment (rlubos) [inline west.yml:59]: I guess you meant `pull/63/head`?

Comment (JakubPegza) [reply west.yml]: That's correct. Fixed

**Review by hubertmis — APPROVED** [review]

File: west.yml [hunk 2]
```diff
@@ -56,7 +56,7 @@ manifest:
       revision: f1fa8241f8786198ba41155413243de36ed878a5
       path: modules/hal/infineon
     - name: hal_nordic
-      revision: 6038f912fd34fdd46dbf017057908f979e298341
+      revision: pull/63/head
```
Comment (rlubos) [inline west.yml:59]: Please update SHA: `844f117c70ffdd5c119b33731ab4fd285e524aee`

Comment (JakubPegza) [reply west.yml]: Done

**Review by rlubos — APPROVED** [review]

File: modules/Kconfig.nordic [hunk 3]
```diff
@@ -69,6 +69,24 @@ config NRF_802154_CCA_CORR_LIMIT
...
+    default 16
+    help
+        Number of slots containing extended addresses of nodes for which pending data is stored
+
+config NRF_802154_RX_BUFFERS
+    int "nRF 802.15.4 receive buffers"
+    default 16
+    help
+        Number of buffers in nRF 802.15.4 driver receive queue.
+
```
Comment (jukkar) [inline modules/Kconfig.nordic:89]: Minor nit, but indentation seems to be slightly off in the added lines (compared to old ones).

Comment (JakubPegza) [reply modules/Kconfig.nordic]: Old lines used tabs instead of spaces, aligned that in added lines.

**Review by carlescufi — APPROVED** [review]

File: west.yml [hunk 4]
```diff
@@ -56,7 +56,7 @@ manifest:
       revision: f1fa8241f8786198ba41155413243de36ed878a5
       path: modules/hal/infineon
     - name: hal_nordic
-      revision: 6038f912fd34fdd46dbf017057908f979e298341
+      revision: 844f117c70ffdd5c119b33731ab4fd285e524aee
```
Comment (carlescufi) [inline west.yml:59]: SHA checked

**Pull request merged**